Feature: Add support for function-based completion in 'complete' option (#17065)
vim/vim: runtime/doc/options.txt

		|i_CTRL-X_CTRL-D|
	]	tag completion
	t	same as "]"
	f{func}	call the function {func}. Multiple "f" flags may be specified.
		Refer to |complete-functions| for details on how the function
		is invoked and what it should return.  The value can be the
		name of a function or a |Funcref|.  For |Funcref| values,
		spaces must be escaped with a backslash ('\'), and commas with

> 		double backslashes ('\\').

Reviewer: ```suggestion
		double backslashes ('\\') (see |option-backslash|).
```

> 		If the {func} result includes 'refresh' set to 'always',

Reviewer: ```suggestion
		If the Dict returned by the {func} includes {'refresh': 'always'},
```

> 		If the Dict returned by the {func} includes {'refresh': 'always'},

Reviewer 2: ```suggestion
		If the Dict returned by the {func} includes {"refresh": "always"},
```
